### DISCUSSION TITLE
Typescript with VS Code debug issue

### DISCUSSION THREAD

**Discussion by AaronPorts** [avajs/ava#3278] [Q&A] [answered]
It seems like debugging in VSCode is not working with Typescript in my case. I'm working in a monorepo, Node v21.4.0, TS 5.3.3, Ava 6.0.1. 

ava.config.js:
```js
  failFast: false,
  timeout: "30s",
  files: ["tests/**/*.test.ts"],
  extensions: { ts: "module" },
  workerThreads: false,
  require: ["dotenv/config"],
```
I'm using tsx and JavaScript Debug Terminal: `cross-env NODE_OPTIONS=--import=tsx DOTENV_CONFIG_PATH=./.development.env ava` , but the breakpoints are not being triggered.

**Comment by novemberborn** [comment]
There are a lot of variables there. My hunch is the `tsx` loader gets in the way.

**Comment by AaronPorts** [comment, marked as answer]
Ok, debugging seems to work when I use `nodeArguments: ["--import=tsx"]` but fails when `NODE_OPTIONS: "--import=tsx"`
